src/tls/openssl/tls.c

```diff
@@ -1597,6 +1627,20 @@ void tls_disable_verify_server(struct tls *tls)
 }
 
 
+/**
+ * Enables SIP TLS client verifications for following requests
+ *
+ * @param tls     TLS Object
+ */
+void tls_enable_verify_client(struct tls *tls)
+{
+	if (!tls)
+		return;
+
+	tls->verify_client = true;
```

Comment:
Perhaps move the `bool` to the API so that the feature can be turn on and off at any time ?

Reply:
I am unsure where I would move it. We have to call `SSL_set_verify` on the SSL object. Ideally, shortly after initializing it. For incoming connections, this happens in the `tcp_connect_handler` in `transp.c` and in this context we don't have much information except for the `sip_transport` which contains the `tls` struct.

This closely mirrors the approach of `verify_server`.

Maybe I misunderstood, could you specify where exactly I could move the `bool`? Or should I add `tls_disable_verify_client` to be able to disable the feature?

Reply:
I think @alfredh is looking for something like this:

```c
void tls_enable_verify_client(struct tls *tls, bool enable) {
	if (!tls)
		return;

	tls->verify_client = enable;
}
```

Reply:
Ooh, okay. Done.